src/restconfig/src/main/java/org/geoserver/catalog/rest/CoverageResource.java

```diff
@@ -150,8 +151,9 @@ protected void handleObjectPut(Object object) throws Exception {
         
         CoverageStoreInfo cs = catalog.getCoverageStoreByName(workspace, coveragestore);
         CoverageInfo original = catalog.getCoverageByCoverageStore( cs,  coverage );
-        new CatalogBuilder(catalog).updateCoverage(original,c);
         calculateOptionalFields(c, original);
+        
+        new CatalogBuilder(catalog).updateCoverage(original,c);
```

Comment:
Are you really just moving calculateOptionalFields before updating the coverage?

If you add this "calculateOptionalFields" to AbstractCatalogResource API (ie as part of the resource lifecycle) can CatalogBuilder make use of the method to make this occur in a consistent order?